test/shared/src/main/scala/zio/test/mock/Method.scala

```diff
@@ -31,3 +58,15 @@ trait Method[+A, -B] {
     }
   }
 }
+
+object Method {
+
+  /**
+   * Evidence type `A` is not equal to type `B`.
+   */
+  trait =!=[A, B]
```

Comment:
Maybe add some `implicitNotFound` or `implicitAmbiguous` annotations here? Right now if this doesn't exist I think the user is going to get a somewhat confusing message about an implicit he or she probably doesn't even realize exists.

Reply:
Good idea! Definitely will do!

Reply:
@adamgfraser turns out `implicitAmbiguous` is only available since scala 2.12, so I cannot use it :(

The `implicitNotFound` does not apply here, as the usage is in `Method` trait and implicit for all `A, B` where `A != B` is also defined in `Method` companion object.

Reply:
Good point about `implicitAmbiguous`. I was playing with `implicitNotFound` and in 2.11 the `implicitNotFound` warning will be triggered if the implicit can't be found because it is ambiguous. So you could have a 2.11 and a 2.12+ version of the file where one uses `implicitAmbiguous` and the other uses `implicitNotFound`. But I know that is a bit of a hassle, so up to you whether you think it is worth it.

Reply:
Alright, I think I got this. Thanks, just learned something new :)